File: src/test/java/com/flexpoker/game/command/aggregate/CreateNewGameTest.java
package com.flexpoker.game.command.aggregate;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.UUID;

import org.junit.jupiter.api.Test;

import com.flexpoker.game.command.commands.CreateGameCommand;
import com.flexpoker.game.command.events.GameCreatedEvent;

public class CreateNewGameTest {

    @Test
    public void testCreateNewGameSuccess() {
        var createGameCommand = new CreateGameCommand("test", 2, 2, UUID.randomUUID(), 10, 20);
        var game = new DefaultGameFactory().createNew(createGameCommand);

        assertEquals(1, game.fetchAppliedEvents().size());
        assertEquals(1, game.fetchNewEvents().size());
        assertEquals(GameCreatedEvent.class, game.fetchNewEvents().get(0).getClass());

        var gameCreatedEvent = (GameCreatedEvent) game.fetchNewEvents().get(0);
        assertNotNull(gameCreatedEvent.getAggregateId());
    }

}

File: src/test/kotlin/com/flexpoker/game/command/aggregate/CreateNewGameTest.kt
package com.flexpoker.game.command.aggregate

import com.flexpoker.game.command.aggregate.eventproducers.createGame
import com.flexpoker.game.command.commands.CreateGameCommand
import com.flexpoker.game.command.events.GameCreatedEvent
import com.flexpoker.test.util.UnitTestClass
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import java.util.UUID

@UnitTestClass
class CreateNewGameTest {

    @Test
    fun testCreateNewGameSuccess() {
        val playerId = UUID.randomUUID()
        val command = CreateGameCommand("test", 2, 2, playerId,
            10, 20)
        val newEvents = createGame(command.gameName, command.numberOfPlayers, command.numberOfPlayersPerTable,
            command.createdByPlayerId, command.numberOfMinutesBetweenBlindLevels, command.numberOfSecondsForActionOnTimer)
        assertEquals(1, newEvents.size)
        val gameCreatedEvent = newEvents[0] as GameCreatedEvent
        assertEquals("test", gameCreatedEvent.gameName)
        assertEquals(2, gameCreatedEvent.numberOfPlayers)
        assertEquals(2, gameCreatedEvent.numberOfPlayersPerTable)
        assertEquals(playerId, gameCreatedEvent.createdByPlayerId)
        assertEquals(10, gameCreatedEvent.numberOfMinutesBetweenBlindLevels)
        assertEquals(20, gameCreatedEvent.numberOfSecondsForActionOnTimer)
        assertNotNull(gameCreatedEvent.gameId)
        assertNotNull(gameCreatedEvent.aggregateId)
    }

}